eip-0018.md

```diff
@@ -0,0 +1,387 @@
+# ErgoFund contracts
+
+* Author: code-for-uss, kushti
+* Status: Proposed
+* Created: 5-Nov-2021
+* License: CC0
+* Forking: not needed 
+
+## Motivation 
+
+Collecting funds in different contexts is needed to build common infrastructure and applications in the Ergo ecosystem. Therefore, this EIP is proposing contracts and standardized box formats for announcing crowdfunding campaigns and collecting funds.
+
+## Overall Design
+
+The design of the crowdfunding contracts and box templates below is centered by a seamless on-chain and off-chain code built on top of the Ergo Playgrounds, [ErgoFund scenarios via playground](https://scastie.scala-lang.org/TWlB0hncT5SdYcTax3Wj5A). 
+
+There is an old version of ErgoFund eip [here](#old-ergofund-contracts), prepared by Kushti.
+
+## Tokens
+
+The ErgoFund has the following types of tokens.
+
+| Token | Issued quantity | purpose | where stored |
+|---|---|---|---|
+|ControlBoxNFT | 1 | Identify control box | Any  |
+|SellNFT | 1 | Identify sell box | Sell box |
+|Registration tokens | 100 million | Identify each campaign box | Sell box <br /> Campaign boxes |
+|Campaign participant tokens | 100 million per campaign | Identify campaign and each pledge box| Capmaign box <br /> pledge boxes |
+
+## Boxes
+
+There are a total of 3 contracts, and each corresponds to a box-type + control box.
+
+| Box | Quantity | Tokens | Additional registers used | Purpose | Spending transactions |
+|---|---|---|---|---|---|
+|Control | 1 | ControlBoxNFT | R4: Price (Long), <br /> R5: address[script] (SigmaProp) | Control price and address to pay for campaign registration | <div align="center">-</div> | 
+|Sell| 1 | SellNFT <br /> Registration tokens | <div align="center">-</div> | Register a new campaign | registerCampaign, <br/> returnRegistrationToken |
+|Campaign | 0 - 100 million | Registration token <br /> Campaign participant tokens | R4: Campaign description (Coll[Byte]), <br /> R5: Public key of project owner (SigmaProp), <br /> R6: Campaign deadline (Int), <br /> R7: minToRaise (Long), <br /> R8 (optional): campaign Proposal CID (Coll[Byte]) | Store campaign data and collect funds | Donate to campaign, <br /> Withdraw funds, <br /> Refund donates, <br /> Return registrationToken |
+|pledge | 0 - 100 million (per campaign) | Campaign participant tokens | R4: Donate value (Long), <br /> R5: Public key of backer (SigmaProp) | Store donate value and Backer public key | Refund donate (fundraising failure) |
+
+## Contracts
+
+### Pledge Contract
+
+This box can only be spent if the first input box, which is the campaign box related to this pledge. Which means the campaign has failed.
+(Conditions: fund collected for the project is less than the defined minToRaise and the height of the network exceeds the fundRaisingDeadline), also, in this case, the fee of funds' returning will be deducted from the backer's fund.<br />
```

Comment:
It can be considered to let the campaign owner finish the campaign if funds are raised even before the deadline.